### PR TITLE
Uifc 89 canche filter queries

### PR DESCRIPTION
**Pull request by elsenhans**

### CODE REVIEW

File: src/components/MetadataCollections/MetadataCollections.js [hunk 1]
```diff
@@ -192,15 +203,53 @@ class MetadataCollections extends React.Component {
...
   render() {
     const { intl, queryGetter, querySetter, onSelectRow, selectedRecordId, collection } = this.props;
     const count = collection ? collection.totalCount() : 0;
 
     return (
       <div data-test-collections>
         <SearchAndSortQuery
-          initialFilterState={{ metadataAvailable: ['yes'] }}
-          initialSearchState={{ query: '' }}
+          // initialFilterState={{ metadataAvailable: ['yes'] }}
```
Comment (rchr) [inline src/components/MetadataCollections/MetadataCollections.js]: Please remove commented out code. Git still remembers this change.

File: src/components/MetadataCollections/MetadataCollections.js [hunk 2]
```diff
@@ -192,15 +203,53 @@ class MetadataCollections extends React.Component {
...
+
+    this.setState({
+      storedFilter: defaultFilter,
+      storedSearchString: defaultSearchString,
+    });
+
+    return (this.props.history.push(`${urls.collections()}?filters=${defaultFilter.string}`));
+  }
+
+  handleClearSearch(getSearchHandlers, onSubmitSearch, searchValue) {
```
Comment (rchr) [inline src/components/MetadataCollections/MetadataCollections.js]: The function does not use the parameter _onSubmitSearch_. Can be removed.

File: src/components/MetadataCollections/MetadataCollections.js [hunk 3]
```diff
@@ -253,9 +306,10 @@ class MetadataCollections extends React.Component {
                         </div>
                         <Button
                           buttonStyle="none"
-                          disabled={disableReset()}
+                          // disabled={disableReset()}
                           id="clickable-reset-all"
-                          onClick={resetAll}
+                          // onClick={resetAll}
+                          onClick={() => this.resetAll(getFilterHandlers(), getSearchHandlers(), resetAll)}
```
Comment (rchr) [inline src/components/MetadataCollections/MetadataCollections.js]: Here this.resetAll() is called with three parameters, but its specification only has two.

File: src/components/MetadataCollections/MetadataCollections.js [hunk 4]
```diff
@@ -238,12 +290,13 @@ class MetadataCollections extends React.Component {
...
                             onChange={getSearchHandlers().query}
-                            onClear={getSearchHandlers().reset}
+                            // onClear={getSearchHandlers().reset}
+                            onClear={() => this.handleClearSearch(getSearchHandlers(), onSubmitSearch(), searchValue)}
                             value={searchValue.query}
                           />
                           <Button
                             buttonStyle="primary"
-                            disabled={!searchValue.query || searchValue.query === ''}
+                            // disabled={!searchValue.query || searchValue.query === ''}
```
Comment (rchr) [inline src/components/MetadataCollections/MetadataCollections.js]: I think it is good practice, if buttons are disabled, if their use has no impact (e.g. the form did not change).

File: src/components/MetadataCollections/MetadataCollections.js (hunk 2 above)
Comment (elsenhans) [reply src/components/MetadataCollections/MetadataCollections.js]: _onSubmitSearch_ is necessary, its in the return statement now

File: src/components/MetadataCollections/MetadataCollections.js (hunk 1 above)
Comment (elsenhans) [reply src/components/MetadataCollections/MetadataCollections.js]: code, which is commented out has been deleted

File: src/components/MetadataCollections/MetadataCollections.js (hunk 3 above)
Comment (elsenhans) [reply src/components/MetadataCollections/MetadataCollections.js]: the unused parameter has been deleted

File: src/components/MetadataCollections/MetadataCollections.js (hunk 4 above)
Comment (rchr) [reply src/components/MetadataCollections/MetadataCollections.js]: Will be fixed in different ticket.

File: src/components/MetadataCollections/MetadataCollections.js (hunk 3 above)
Comment (rchr) [reply src/components/MetadataCollections/MetadataCollections.js]: You forgot remove the parameter here in MetadataCollections.js (you did remove it just in MetadataSources.js).

**Pull request merged**